app/ui/src/main/java/de/mm20/launcher2/ui/launcher/widgets/clock/clocks/OrbitClockVM.kt

```diff
@@ -0,0 +1,21 @@
+package de.mm20.launcher2.ui.launcher.widgets.clock.clocks
+
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+
+class OrbitClockVM : ViewModel(), KoinComponent {
```

Comment:
This is terribly inefficient. `viewModelScope` is only cancelled when the ViewModel is cleared, meaning this will continue to update even when the launcher is not in foreground. You should move this to a `LaunchedEffect` inside the composable and wrap it in `lifecycle.repeatOnLifecycle(...) { ... }` to make sure that this is paused and restarted when the launcher is paused / resumed.

Reply:
I've changed it, but I am not sure if that is what you meant. I found similar logic in ProvideCurrentTime.kt.

Still, how can we make sure the coroutine within the Orbit-Clock's `repeatOnLifecycle` is canceled when a different clock-layout is chosen? The code in ProvideCurrentTime could be changed to update every second, but I think this is somewhat wasteful for clock layouts that don't need second accuracy.

Reply:
`LaunchedEffect` (and its children) are canceled when it leaves the composition.

Reply:
> The code in ProvideCurrentTime could be changed to update every second, but I think this is somewhat wasteful for clock layouts that don't need second accuracy.

Another approach is used for [dynamic clock icons](https://github.com/MM2-0/Kvaesitso/blob/0adef99d4e65293af1f558d150fc59aa69a5333e/app/ui/src/main/java/de/mm20/launcher2/ui/component/ShapedLauncherIcon.kt#L334):

There, the seconds hand is moved by a 60s animation which is restarted whenever the time (provided by LocalTime) is updated.

Reply:
> There, the seconds hand is moved by a 60s animation which is restarted whenever the time (provided by LocalTime) is updated.

I've thought about introducing another orbit layout like this too, with smooth animations and orbit paths similar to these [typical loading circles](https://dm0qx8t0i9gc9.cloudfront.net/thumbnails/video/IevwMJs/loading-circle-gradient-w-alpha-25fps_s6zya32g_thumbnail-1080_01.png ), for them to look like comet-trails. Wasn't sure about this though, if it would fit into the launcher's design so I figured to change the current one first and then think about it again. For this, the smooth animation would be pretty handy, yes. What do you think?